pkg/controller/secret/secret_controller.go

```diff
@@ -169,12 +173,41 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 		// valid userData secret already exists
 		return reconcile.Result{}, nil
 	} else {
-		// secret is updated
+		// userdata secret data does not match what is expected
+		// Mark nodes configured with the previous private key for deletion
+		signer, err := signer.Create(privateKey)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "error creating signer from private key")
+		}
+		nodes := &core.NodeList{}
+		err = r.client.List(context.TODO(), nodes, client.HasLabels{nodeconfig.WindowsOSLabel})
+		if err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "error getting node list")
+		}
+		expectedPubKeyAnno := nodeconfig.CreatePubKeyHashAnnotation(signer.PublicKey())
```

Comment:
This is only being ran when the userdata/privatekey has been changed so the hash will be different than what it was.

Reply:
nit: I thought we'd run it on every reconcile even if there is version change. Agreed it may be short circuited but why to do it if we know it hasn't changed.

Reply:
There are two ways to get to this, either the private key has changed or the userdata has changed. The userdata will only be changed if either the private key has changed or someone is tampering with the secret. I dont think we should consider tampering with the secret something that will happen in a normal cluster, and I'd prefer to keep the code here simpler.